appsec-kit-starter/src/test/java/com/vaadin/appsec/service/AppSecDevToolsPluginTest.java

```diff
@@ -0,0 +1,156 @@
+        when(appSecServiceInstance.getNewVulnerabilities())
+                .thenReturn(Collections.emptyList());
+        when(appSecServiceInstance.addScanEventListener(any()))
+                .thenReturn(() -> {
+                });
+
+        appSecDevToolsPlugin.handleConnect(devToolsInterface);
+
+        verify(appSecServiceInstance, times(1)).addScanEventListener(any());
+        assertEquals(1, appSecDevToolsPlugin.scanEventRegistrations.size());
+    }
+
+    @Test
+    public void handleConnect_doesntAddScanEventListener_ifAlreadyAdded() {
+        when(appSecServiceInstance.getNewVulnerabilities())
+                .thenReturn(Collections.emptyList());
+        appSecDevToolsPlugin.scanEventRegistrations.put(devToolsInterface,
+                () -> {
+                });
+
+        appSecDevToolsPlugin.handleConnect(devToolsInterface);
+
+        verify(appSecServiceInstance, never()).addScanEventListener(any());
+        assertEquals(1, appSecDevToolsPlugin.scanEventRegistrations.size());
+    }
+
+    @Test
+    public void scanEventListener_sendsScanResult() {
+        var appSecScanEvent = mock(AppSecScanEvent.class);
+        when(appSecScanEvent.getNewVulnerabilities())
+                .thenReturn(Collections.emptyList());
+        when(appSecServiceInstance.getNewVulnerabilities())
+                .thenReturn(Collections.emptyList());
+
+        appSecDevToolsPlugin.handleConnect(devToolsInterface);
+
+        verify(appSecServiceInstance)
+                .addScanEventListener(appSecScanEventListenerCaptor.capture());
+        var appSecScanEventListener = appSecScanEventListenerCaptor.getValue();
+        appSecScanEventListener.scanCompleted(appSecScanEvent);
+
+        verify(devToolsInterface, times(3)).send(anyString(), any());
+    }
+
+    @Test
+    public void handleMessage_returnsTrue() {
+        var result = appSecDevToolsPlugin.handleMessage("command",
+                Json.createObject(), devToolsInterface);
+        assertTrue(result);
+    }
+
+    @Test
+    public void handleDisconnect_removesScanEventListener() {
+        var registration = mock(Registration.class);
+        appSecDevToolsPlugin.scanEventRegistrations.put(devToolsInterface,
+                registration);
+
+        appSecDevToolsPlugin.handleDisconnect(devToolsInterface);
+
+        verify(registration, times(1)).remove();
+        assertEquals(0, appSecDevToolsPlugin.scanEventRegistrations.size());
+    }
+
+    @Test
+    public void handleDisconnect_doesntRemoveScanEventListener_ifAlreadyRemoved() {
+        var devToolsInterface1 = mock(DevToolsInterface.class);
+        var registration = mock(Registration.class);
+        appSecDevToolsPlugin.scanEventRegistrations.put(devToolsInterface1,
+                registration);
+
+        appSecDevToolsPlugin.handleDisconnect(devToolsInterface);
+
+        verify(registration, never()).remove();
+        assertEquals(1, appSecDevToolsPlugin.scanEventRegistrations.size());
```

Comment:
If I understand correctly the purpose of the test, this is purely internal logic. To properly test this we should connect a different `DevToolsInterface` and verify it still receives messages after the other one is disconnected.

Reply:
Removed this test and added the suggested verification in the `handleDisconnect_removesScanEventListener` test.